File: LeetCode/Note_0520_Detect_Capital_E/Note_0520_Detect_Capital_E.py
import re

class Solution(object):
    def detectCaptial(self, word: str) -> bool:
        return re.fullmatch(r"[A-Z]*|.[a-z]*", word)

if __name__ == '__main__':
    print("template")
    solution = Solution()
    
    print("OutPut: ", solution.detectCaptial("word"), ", Shouls be True");
    print("OutPut: ", solution.detectCaptial("Google"), ", Shouls be True");
    print("OutPut: ", solution.detectCaptial("USA"), ", Shouls be True");
    print("OutPut: ", solution.detectCaptial("GoogLe"),", Shouls be False");

    print(re.fullmatch(r"[A-Z]*", "A"))
    print(re.fullmatch(r"[A-Z]*", "aaaa"))
    print(re.fullmatch(r"[A-Z]*", "AaaA"))
    print(re.fullmatch(r"[A−Z][a−z]*", "Aaaa"))
# T:O(n)
# S:O(n)

File: LeetCode/Note_0520_Detect_Capital_E/Note_0520_Detect_Capital_E.java


public class Note_0520_Detect_Capital_E {

    // public Note_0520_Detect_Capital_E() {

    // }

    public boolean detectCapital(String word) {

        return word.matches("[A-Z]*|.[a-z]*");
    }

    public static void main(String []args) {
        System.out.println("Detect Capital!"); 
        Note_0520_Detect_Capital_E solution = new Note_0520_Detect_Capital_E();
        System.out.println("OutPut: " + solution.detectCapital("word") + ", Shouls be True");
        System.out.println("OutPut: " + solution.detectCapital("Google") + ", Shouls be True");
        System.out.println("OutPut: " + solution.detectCapital("USA") + ", Shouls be True");
        System.out.println("OutPut: " + solution.detectCapital("GoogLe") + ", Shouls be False");
    }
}